### PR TITLE
CNF-23406: Add unit tests for pkg/utils

### PR DESCRIPTION
**Pull request by sebrandon1**
## Summary
- Add 27 unit tests for `pkg/utils` which previously had zero test coverage across ~440 LOC
- Tests cover 10 functions: `parseImageStreamTagString`, `getPodDefinition`, `getNamespaceDefinition`, `ListNodes`, `WriteFile`, `GetPlatformType`, `GetControlPlaneTopology`, `GetClusterVersion`, `IsIPv6Enabled`, `DeletePod`
- Uses Ginkgo v2 / Gomega with `controller-runtime/pkg/client/fake` for k8s client faking

## Test plan
- [x] `make test` passes (all unit tests green)
- [x] `make lint` passes clean
- [ ] CI checks pass

Jira: [CNF-23406](https://redhat.atlassian.net/browse/CNF-23406)

### CODE REVIEW

File: pkg/utils/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+)
+
+var testScheme *runtime.Scheme
+
+func init() {
```
Comment (oribon) [inline pkg/utils/utils_test.go]: this can be BeforeSuite I think

File: pkg/utils/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+func newFakeUtils(objs ...runtime.Object) *utils {
+	fakeClient := fake.NewClientBuilder().
+		WithScheme(testScheme).
+		WithRuntimeObjects(objs...).
+		Build()
+
+	return &utils{&client.ClientSet{Client: fakeClient}}
+}
+
+var _ = Describe("parseImageStreamTagString", func() {
```
Comment (oribon) [inline pkg/utils/utils_test.go]: this can be converted to a DescribeTable

File: pkg/utils/utils_test.go [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+	It("returns an error when ClusterVersion CR is missing", func() {
+		u := newFakeUtils()
+		_, err := u.GetClusterVersion()
+		Expect(err).To(HaveOccurred())
+	})
+})
+
+var _ = Describe("IsIPv6Enabled", func() {
+	It("returns false for IPv4-only cluster", func() {
```
Comment (oribon) [inline pkg/utils/utils_test.go]: same, can be a table
